src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -39,58 +37,71 @@ import org.gradle.kotlin.dsl.withType
 @Suppress("UnstableApiUsage")
 class NexusPublishPlugin : Plugin<Project> {
 
-    companion object {
-        private val serverUrlToStagingRepoUrl = ConcurrentHashMap<URI, URI>()
-    }
-
     override fun apply(project: Project) {
-        project.pluginManager.apply(MavenPublishPlugin::class)
-
-        project.gradle.addBuildListener(object : BuildAdapter() {
-            override fun buildFinished(result: BuildResult) {
-                serverUrlToStagingRepoUrl.clear()
-            }
-        })
+        require(project == project.rootProject) {
+            "Plugin must be applied to the root project but was applied to ${project.path}"
+        }
 
         val extension = project.extensions.create<NexusPublishExtension>(NexusPublishExtension.NAME, project)
+        configureNexusTasks(project, extension)
+        configurePublishingForAllProjects(project, extension)
+    }
 
+    private fun configureNexusTasks(project: Project, extension: NexusPublishExtension) {
```

Comment:
Have you evaluated `private fun Project.configureNexusTasks(extension: NexusPublishExtension) {` so the code looks closer to the regular `build.gradle.kts` (e.g. without duplication of `project.`)?

Reply:
I don't think it's worth it in this case.